node/client/src/lib.rs

```diff
@@ -214,21 +222,21 @@ impl Client {
         let prev_shard_header = prev_shard_block.header();
         let apply_state = ApplyState {
             root: prev_shard_header.body.merkle_root_state,
-            block_index: prev_header.body.index,
+            block_index: prev_header.body.index + 1,
             parent_block_hash: parent_hash,
             shard_id: shard_block.body.header.shard_id,
         };
         let apply_result = self.runtime.write().check(
             &apply_state,
-            &prev_shard_block.body.new_receipts,
+            &[],
```

Comment:
why are previous receipts not needed in `Runtime.check`. it's unclear from `check`. it would be better if we made them optional and documented when they were required in which case, or create separate signatures for checking imported blocks and produced blocks.

Reply:
I am going to change runtime soon and will rewrite `check` and `apply`